cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -5049,6 +5067,50 @@ if(ARROW_S3)
   endif()
 endif()
 
+# ----------------------------------------------------------------------
+# Azure SDK for C++
+
+function(build_azure_sdk)
+  message(STATUS "Building Azure SDK for C++ from source")
+  fetchcontent_declare(azure_sdk
+                       URL ${ARROW_AZURE_SDK_URL}
+                       URL_HASH "SHA256=${ARROW_AZURE_SDK_BUILD_SHA256_CHECKSUM}")
+  fetchcontent_getproperties(azure_sdk)
+  if(NOT azure_sdk_POPULATED)
+    fetchcontent_populate(azure_sdk)
+    set(BUILD_PERFORMANCE_TESTS FALSE)
+    set(BUILD_SAMPLES FALSE)
+    set(BUILD_TESTING FALSE)
+    set(BUILD_WINDOWS_UWP TRUE)
+    set(CMAKE_EXPORT_NO_PACKAGE_REGISTRY TRUE)
+    set(DISABLE_AZURE_CORE_OPENTELEMETRY TRUE)
+    set(ENV{AZURE_SDK_DISABLE_AUTO_VCPKG} TRUE)
+    add_subdirectory(${azure_sdk_SOURCE_DIR} ${azure_sdk_BINARY_DIR} EXCLUDE_FROM_ALL)
```

Comment:
Is there a reason not to use `fetchcontent_makeavailable` in favor of manually populating?

Reply:
`add_subdirectory` was recommended in the Azure docs https://github.com/Azure/azure-sdk-for-cpp/blob/main/README.md#2-define-cmake-fetch-content . Beyond that I'm not aware of any reason.

Reply:
It might be to add `EXCLUDE_FROM_ALL`? But I don't think this is necessary here. So you can just remove the if etc. and replace add_subdir with `fetchcontent_makeavailable(azure_sdk)`. You just have to set the variables before that call.

Reply:
done